### ISSUE TITLE
Selected good first issues for new contributors

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#1147]
A number of good issues to start working on for the 2021 scipy sprint.
Contributions to documentation are especially appreciated.

### introductory and user testing

* work through the [sktime tutorials](https://www.sktime.org/en/latest/tutorials.html) and record feedback
* work through user testing sheets for [forecasting](https://github.com/sktime/sktime-workshops/blob/main/sktime_worksheet_forecasting.ipynb) or [time series classification](https://github.com/sktime/sktime-workshops/blob/main/sktime_worksheet_TSC.ipynb), record feedback and send us your user testing sheet with code (e.g., attach to this issue)
* open an issue with a time series related algorithm that you would like to see implemented or interfaced in `sktime`
* work through the [installation instructions](https://www.sktime.org/en/latest/installation.html) to install a development version of `sktime`. Record carefully things that are unclear or don't work as described, return the full feedback to us (e.g., in an issue).

### contributors new to open source

* documentation: many code files are missing module and class/function docstrings. Pick a file and start adding these. A good starting point are estimator algorithms, these should follow the general documentation style outlined in #1148 or the [extension templates](https://github.com/alan-turing-institute/sktime/tree/main/extension_templates). If you start, post the name of the estimator you are working on in #1148; when you open a PR, refer to the issue.
* documentation: many code files are missing examples. Pick a file and start adding these. a good starting point are estimator algorithms, and time series classifiers, see #932 and #1148

### small-to-medium technical writing tasks

* the [user guide](https://www.sktime.org/en/latest/user_guide.html) is incomplete. Help is appreciated - the best approach may be converting the [tutorials](https://www.sktime.org/en/latest/tutorials.html) in a well-written guide.
* the [tutorial for time series classification](https://www.sktime.org/en/latest/tutorials.html) should be re-factored, in the style of the forecasting tutorial
* extension templates for time series regression and time series annotation should be written. The current [extension templates](https://github.com/alan-turing-institute/sktime/tree/main/extension_templates) can be used as an extension template template.

### small-to-medium python/coding tasks

* help refactor the time series classifiers to the new interface, as outlined in the extension template, see #1146 for a recipe. Pick one classifier and post on the issue which.
* Pick an algorithm to implement or interface, according to the [extension templates](https://github.com/alan-turing-institute/sktime/tree/main/extension_templates). Possible choices are forecasters (univariate or multivariate), time series classifiers, time series regressors, annotators, distances, kernels - these can be existing feature requests, or new suggestions. We currently do not recommend transformers as they undergo refactoring.
* the [panel data conversions module](https://github.com/alan-turing-institute/sktime/blob/main/sktime/utils/data_processing.py) is incomplete and a bit unordered. Help with writing docs for format specifications, or contributing converters to unsupported from/to combinations is appreciated.
* implement retrieval functionality for the registry: `all_tags`, or `all_estimators` with ability to sub-set by tags
* implement `get_fitted_params` functionality in estimators where this is missing (currently most of them)

### package maintenance tasks

* help create a conda metapackage #1142 or a comparable solution for installing `sktime` with soft dependencies on `conda`
* refactor test suite: help move default parameter mechanics to individual classes #1098.

### challenging tasks

✨ Your time to shine! ✨ 
Talk to a core dev before starting with these.
Also have a look at the roadmap #228

* python 3.9 support
* `get_report` interface for model inspection #971
* refactor/redesign: forecasting interface for stream update/prediction #982
* refactor/redesign: forecasting interface for prediction intervals #984
* refactor/redesign: test suite #1032, #1097, #1098
* help build/design new experimental module: distances/kernels
* help build/design new experimental module: annotations (segmentation, change point detection, outlier detection)
* help build/design new experimental module: sequence alignment

**Comment by mloning** [comment]
Closing this now but perhaps we should rework this into a template for sprint planning, some of the items will be relevant for future sprints too?

**Comment by fkiraly** [comment]
Why not just open it and remove the scipy reference? Still good for pinning, no?

**Comment by gepitis** [comment]
Hi, thanks for the list of first issues! Where can I send my feedback on the tutorials?

**Comment by aiwalter** [comment]
Hi @gepitis ,
you could open a new issue with ideas of improvement that you have in mind

**Comment by fkiraly** [comment]
>Where can I send my feedback on the tutorials?

I just opened an issue here
https://github.com/alan-turing-institute/sktime/issues/1447
thanks for any feedback!

**Comment by eyjo** [comment]
The links for the sktime website are for the .org domain.

**Comment by fkiraly** [comment]
thanks for pointing out - the domain got swooped up by third parties, it´s now on sktime.net - fixed all the links!

**Comment by priyakumari02** [comment]
Hello @fkiraly , I want to contribute to GSOC'24 , So this org caught my eye.I tried to join the discord channel , but It is showing invalid link

**Comment by fkiraly** [comment]
Thanks for the note, @priyakumari02!
The discord invite and the link to projects have been updated to 2024.

**Comment by iaryangoyal** [comment]
hi @fkiraly ,  I want to contribute to GSOC'24 , can you spot some good first issue?

**Comment by fkiraly** [comment]
Welcome to sktime! You seem to be new to open source, how about this one?
https://github.com/sktime/sktime/issues/3429

**Comment by MMTrooper** [comment]
Hello @fkiraly, I am pretty new to open source. What are some issues you'd recommend I tackle?

**Comment by fkiraly** [comment]
> I am pretty new to open source and maybe late to the participating.

Welcome, and rest assured that it is never too late!

How about #3429 as a starter? It is relatively simple python-wise, but allows you to practice open source workflows, git, PR, reviews, etc.

**Comment by MMTrooper** [comment]
> > I am pretty new to open source and maybe late to the participating.
> 
> Welcome, and rest assured that it is never too late!
> 
> How about #3429 as a starter? It is relatively simple python-wise, but allows you to practice open source workflows, git, PR, reviews, etc.

On it

**Comment by fkiraly** [comment]
@MihirsinhChauhan, thanks! Truth is, there already have been gradual updates over time, so what would be helpful:

* work through the tutorial and check whether all is clear
* check more recent tutorials from conferences/workshops, on new features, in the `github.com/sktime` organization. See whether the new content from there could be distributed across the existing notebooks, or makes sense to add as a new notebook.

**Comment by sohamchatterjee50** [comment]
Hey @fkiraly 
I would love to cotribute to this project. I have professional exeperience as a Fraud Analyst at a FinTech company and as an Applied Scientist at Amazon.
1. I can work for 16 hrs per week.
2. We have some research courses on Deep Learning and Foundation Models.

I would love to contribute to the Foundation Models project, which will be an enriching reseach experience for me.

**Comment by fkiraly** [comment]
Welcome to sktime, @tunglinwood, @sohamchatterjee50.
Consider applying for mentoring: https://github.com/sktime/mentoring

Also, you can look on discord in the workstream channels for your preferred topics.

**Comment by fkiraly** [comment]
> I implemented the command line `pydocstyle sktime/ --config=setup.cfg | grep ./ | cut -d ':' -f1 | uniq` but nothing really comes into information of docstrings needed to be complete. Do I miss something here?

@tunglinwood, not sure, can you kindly explain what you are trying to do?

**Comment by bhorsujal** [comment]
Hello @fkiraly, I am an open source enthusiast and I tend to use open source softwares as much as possible, but have faced a lot of issues while contributing to open source projects. This project gave me a hope that maybe I could contribute to open source after all.
If possible, Can you please suggest me an issue to get started with? I am ready to devote a lot of time if it's for open source. But getting a good start seems like an issue for me.

**Comment by KeshavSingh2703** [comment]
Hey there, I would like to take this issue

**Comment by kaifshaheemj** [comment]
Hello @fkiraly, I am interested in contributing to GSOC'25. I have knowledge of Time Series and would like to know if you can suggest some good first issues to get started.

**Comment by PrinceMaloo** [comment]
Hey @fkiraly, I’m interested in contributing to sktime for GSoC 2025. I have experience with research publications and internships, but I'm new to open-source contributions. Could you suggest some good first issues to help me get started?

**Comment by fkiraly** [comment]
Sure! There is a list of good first issues to get started at the top of this issue, have a look!

Let us know if you need further guidance, and feel free to chat with the developers on discord (link also above).

**Comment by ire-mide1** [comment]
Hello @fkiraly , I am Iremide and I am interested in contributing to sktime for GSoC2025. Please I can't seem to find good first issues in the summary issue link abobe. Where they tag as sometime else?

**Comment by fkiraly** [comment]
> Is there any plans to have a virtual office hour for interested participants about the projects focus this year?

Yes, if we get accepted to GSoC as an organization, we will hold an info session soon after, which will be announced on the discord.

**Comment by prav-kotte1** [comment]
Hi! I'd like to contribute to this project and take on some small-to-medium tasks. Specifically, I'm interested in:

- Refactoring the time series classification tutorial to match the forecasting tutorial style
- Adding an extension template for time series regression
- Exploring topics from the sktime workshops and contributing to documentation/tutorials
- Working on [ENH] issues like ensuring test parameter sets (#3429) or implementing _evaluate_by_index for a metric (#4304)
- Trying to reproduce and investigate bugs from the [bugfixing board](https://github.com/sktime/sktime/projects/18)

Could you please assign an issue or suggest where I should begin? Thanks!

**Comment by SEBBBOL1** [comment]
Hello

**Comment by KALYANI2309K3** [comment]
Hi @fkiraly and sktime maintainers,
I’d like to work on the following task from this issue:
✅ Add a Jupyter notebook example and unit test for the ExponentialSmoothing estimator (forecasting demo).
Is that all right?

**Comment by fkiraly** [comment]
@KALYANI2309K3, where do you get this notebook task from? Not sure if this is still open.

**Comment by Arpithajain26** [comment]
Hi, @fkiraly I’m a participant in GSSoC 2025 and I would like to work on this issue.
I have already set up the sktime repository locally and am ready to start.
Could you please assign it to me? Thanks!

**Comment by fkiraly** [comment]
Sure! Since this is an umbrella issue, this will not be assigned to you, instead, go to one of the issues that are "good first issues" (pick one), post there.

**Comment by Krishnasingh020** [comment]
Hi, I’m Krishna singh, an aspiring GSoC 2025 contributor. I’d love to work on this issue. May I take it up?

**Comment by fkiraly** [comment]
sure, please pick a good first issue and post there!

**Comment by Arpithajain26** [comment]
I am new to this contribution can u pls help me with this ? Which to choose

**Comment by priti200** [comment]
Hi @fkiraly 
I’m Priti Gupta, and I’m really excited to start contributing to sktime as I prepare for ESOC 2026.

I’ve set up the project locally and have been reading through the developer guide to understand how everything works. I’d love to begin with a small task so I can learn the contribution workflow properly and submit my first PR.

It would be great if you could point me to a few beginner-friendly issues. I’m happy to help with documentation, adding examples, tests, or anything else that’s currently useful for the project.

Thanks a lot for your guidance. Looking forward to contributing and learning from the community!

**Comment by Joiejoie1** [comment]
I’m Joy, and I’ve just officially submitted my applicant registration form for ESoC 2026 to contribute to sktime!
I'll like to contribute to this repo.

My Background:
I am a Full-Stack Developer with a heavy focus on the cloud-native ecosystem. I’m an active contributor to Meshery (Layer5) within the CNCF, where I've worked on high-visibility projects like the Kanvas-site (Next.js) and ecosystem integrations.

Why sktime?
My domain expertise lies in HealthTech and EduTech. In both fields, time-series data is the 'source of truth', whether it’s monitoring longitudinal patient data or analyzing student learning curves over a semester. I’m excited to bring my engineering background to a project that makes time-series analysis more accessible and robust.

What I bring to the table:

Technical Stack: Deep experience in Python, React/Next.js, and Go.

Systems Thinking: My work with Kubernetes and Meshery means I'm comfortable with complex CI/CD pipelines, package management, and infrastructure-as-code.

Full-Stack Perspective: I love building tools that aren't just powerful under the hood, but also have intuitive, high-performance interfaces.

I’m currently diving into the Batch 1 project list and looking for where my 'Systems & Full-Stack' skills can best support the sktime roadmap whether that's improving core estimators, enhancing the documentation/web-tooling, or optimizing performance.

**Comment by Aditya-54** [comment]
Hey @fkiraly 
I’m Aditya, CS student who has been exploring time series analysis through several of my own projects. I have a strong interest in ML and experience working with various forecasting and modeling libraries. I recently applied for ESOC 2026 and wanted to introduce myself here.

While getting familiar with the codebase, I submitted a small PR (#9452) to fix some documentation typos as a starting point. I’m looking forward to contributing more if there are any “good first issues” or smaller tasks that could use an extra pair of hands, I’d be happy to help.

I would really value the opportunity to contribute to the project and grow through hands-on involvement..

**Comment by AbdullahShheik** [comment]
Hi @fkiraly!
I'm Abdullah Shaikh, a Computer Science student with experience and a strong interest in machine learning. I’m interested in contributing to this repository as part of my preparation for Google Summer of Code 2026.

I’m relatively new to time series, but I’ve gone through the tutorials mentioned above and am eager to learn more while contributing. If possible, I would really appreciate being pointed to or assigned a good first issue so I can get started.
Thank you!

**Comment by PrabhjotKaur023** [comment]
@fkiraly @felipeangelimvieira

Hi! I am Prabhjot Kaur, an MCA (Master of Computer Applications) student applying for GSoC 2026 Project 6: Stream and online support for forecasting and detection tasks.

As the getting started task mentioned in the GSoC project description, I have implemented a dedicated `_update` method for `NaiveForecaster`, which resolves the long-standing `NotImplementedWarning` present in the current codebase (v0.40.1) and addresses the slow `update_predict` performance reported in #2669.

**Changes made:**
- Added `_update` to `NaiveForecaster` in `_naive.py`
- True incremental update without full refit
- Works correctly for all 3 strategies: `last`, `mean`, `drift`
- Handles `update_params=False` correctly
- Handles duplicate index entries after concat
- Added 6 passing tests in `test_naive_update.py` with no warnings

I have submitted this as a draft PR for early feedback before my GSoC proposal submission. Would really appreciate any feedback or suggestions on the implementation approach!

Thank you!

**Comment by Sidra-009** [comment]
Hi @fkiraly!
I am Sidra, a Data Science student specializing in AI/ML. As part of my GSoC 2026 preparation, I have submitted PR #9640 to address pandas deprecation warnings. I am currently drafting my proposal and look forward to contributing more to sktime's core modules. Excited to work with the community!

**Comment by aparajitaks** [comment]
Hi! I'm Aparajita, a second-year BTech CSE (AI/ML) student.

I’m new to open-source and currently exploring time series analysis through sktime. I’ve started going through the tutorials and trying to understand how everything works.

I’m really interested in learning by contributing and getting hands-on experience. Looking forward to picking up some beginner-friendly issues and gradually improving along the way.

Thanks!

**Comment by Bot87Ever** [comment]
Hi!! I'm Umesh,
I’ve been exploring sktime and going through the tutorials to understand the project better.

I’m interested in contributing and would like to start with something small to get familiar with the codebase and workflow. Could you suggest a beginner-friendly issue, possibly around documentation or simple improvements, that would be a good starting point?

Thanks!!!!!

**Comment by NeerajKumar011** [comment]
Hi @fkiraly,

I'm Neeraj kumar, a student applying for ESoC 2026. I have experience with Python and image processing (I implemented PSNR and Entropy in Scilab for FOSSEE 2025).

I'm new to time series but want to start with a small documentation task or adding examples to docstrings. Could you suggest one specific issue to work on?

Thanks!

**Comment by keerthana-ar** [comment]
Hi @fkiraly, 

I'm Keerthana, a third-year CSE student applying for ESoC 2026. I'm interested in contributing to the detection module, as I'm applying for the Embedded AI for Predictive Sensor Systems project.

I've already submitted a PR to sktime-mcp fixing detection task listing. Looking forward to contributing to sktime core as well

Could you point me to a good detection-related issue to work on? I'm comfortable with Python and ML pipelines and looking for something with some substance, not just a docs fix. Happy to implement a new detector, add multivariate detection support, or work on benchmarking/evaluation for detection tasks.

**Comment by Sonika-19** [comment]
Hi @fkiraly,

I’m K L Sonika, a 3rd-year B.Tech CSE student at PES University, interested in contributing to sktime (ESoC 2026).

I’ve been focusing on forecasting and estimator design, and have already:

* built a reduction-based forecasting prototype inspired by sktime: https://github.com/Sonika-19/fm-ts-interface
* submitted a PR: https://github.com/sktime/sktime/pull/10038

My prototype includes a modular pipeline with reduction → strategy → model abstraction, supporting recursive/direct forecasting and a ForecastingHorizon interface.

I’m particularly interested in contributing to:

* forecasting module (estimators / strategies)
* API design and model integration

I’d like to work on a **core functionality task** aligned with the roadmap rather than a small docs fix.

Would appreciate guidance on where my contributions would be most useful.

Thanks!

**Comment by Krishnasingh020** [comment]
Hi @fkiraly 👋

I’m Krishna, a B.Tech CSE student with experience in Python and full-stack development. I’m currently exploring time-series concepts and have started going through the sktime tutorials.
I’d love to contribute to the project and was planning to begin with a beginner-friendly issue (likely around documentation or adding examples to docstrings).
Could you please suggest a specific issue from the current list that would be a good starting point?
Looking forward to contributing

**Comment by ganesh17091** [comment]
Hi @fkiraly 👋,

I'm Ganesh, a B.Tech Electrical Engineering student from India with a strong interest in Python, AI/ML, and data science. I'm currently preparing for GSoC and have been exploring the sktime codebase and tutorials to better understand time series analysis.

I'm new to contributing to sktime but excited to learn and get involved. I'm looking to start with a beginner-friendly issue, preferably around Python, testing, or documentation, to become familiar with the project's workflow.

Looking forward to learning from everyone and contributing to the project. Thanks! 🚀

**Comment by OxBT3J4** [comment]
Hi, I'm a second-year CS student looking to make my first open source contribution. I've checked several "good first issue" tagged items but they all have existing PRs. Could you point me to one that's currently genuinely unclaimed?
